src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -2524,10 +2524,21 @@ private Object performSqueezeBang(RubyString string, Object[] otherStrings) {
             Rope otherRope = RubyStringLibrary.getUncached().getRope(otherStr);
             Encoding enc = checkEncodingNode.executeCheckEncoding(string, otherStr);
             final boolean squeeze[] = new boolean[StringSupport.TRANS_SIZE + 1];
-            StringSupport.TrTables tables = StringSupport.trSetupTable(otherRope, squeeze, null, true, enc);
 
             boolean singlebyte = rope.isSingleByteOptimizable() && otherRope.isSingleByteOptimizable();
 
+            if (otherStrings.length == 1 && otherRope.characterLength() == 1 && singlebyte) {
```

Comment:
It seems better to test `singlebyte` first here, and then checking `otherRope.byteLength` (instead of `otherRope.characterLength()`) is enough.
The characterLength() in some cases might be expensive to compute.